plugins/modules/win_dns_zone.ps1

```diff
@@ -0,0 +1,231 @@
+#!powershell
+
+# Copyright: (c) 2020 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: GPL-3.0-only
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+    options = @{
+        name = @{ type = "str"; required = $true }
+        type = @{ type = "str"; choices = "primary", "secondary", "forwarder", "stub" }
+        replication = @{ type = "str"; choices = "forest", "domain", "legacy", "none" }
+        dynamic_update = @{ type = "str"; choices = "secure", "none", "nonsecureandsecure" }
+        state = @{ type = "str"; choices = "absent", "present"; default = "present" }
+        forwarder_timeout = @{ type = "int" }
+        dns_servers = @{ type = "list"; elements = "str" }
+    }
+    supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+$check_mode = $module.CheckMode
+
+$name = $module.Params.name
+$type = $module.Params.type
+$replication = $module.Params.replication
+$dynamic_update = $module.Params.dynamic_update
+$state = $module.Params.state
+$dns_servers = $module.Params.dns_servers
+$forwarder_timeout = $module.Params.forwarder_timeout
+
+$parms = @{ name = $name }
+
+Function Get-DnsZoneObject {
+    Param([PSObject]$Object)
+    $parms = @{
```

Comment:
Should this contain `$null` values for the ones you are conditionally setting below?

Reply:
Yep - that's intended. This function exists solely for parsing the object we get back from the cmdlet (`Get-DnsServerZone` for example) and returning a nicely formatted Ansible-friendly hashtable/dict.